shell/platform/android/external_view_embedder/external_view_embedder.cc

```diff
@@ -151,7 +151,7 @@ void AndroidExternalViewEmbedder::SubmitFrame(
 
       // Get the intersection rect between the current rect
       // and the platform view rect.
-      joined_rect.intersect(current_view_rect)
+      joined_rect.intersect(current_view_rect);
```

Comment:
This is not quite the same thing as iOS does. Here you are joining all of the sets of rects for each view all into one rect. In pseudo-code:

iOS:
```
  for (each inter-view slice/picture) {
    for (each view behind that slice) {
      query rects that overlay that underlying view
      (potentially) join them all
      intersect the (potentially joined) rects with the underlying view bounds
      inform iOS of that intersection
    }
  }
```

android:
```
  for (each inter-view slice/picture) {
    create a global join rect
    for (each view behind that slice) {
      query rects that overlay that underlying view
      join them all into the global join rect
    }
    inform Android of that intersection
  }
```

The joins are batched differently on Android. An intersection is needed on the join of the rects from a given trip through the inner loop, so you'd essentially need to create a global and a local join rect and operate on them as follows:

android:
```
  for (each inter-view slice/picture) {
    create a global join rect
    for (each view behind that slice) {
      create a local join rect
      query rects that overlay that underlying view
      join them all into the local join rect
      intersect local join rect with the view bounds
      join the local join rect into the global join rect
    }
    inform Android of the intersection in the global join rect
  }
```

Reply:
Oh, great, I got it. As you said, I shouldn't just calculate the intersection of one rectangle here, otherwise I will have a problem when I have multiple Platformviews. I wrote a demo to find this problem（The green rectangle should be above the two red rectangles）

![image](https://user-images.githubusercontent.com/40540394/208613007-0f701160-415d-4c5c-8d90-011dc047c888.png)

Reply:
After making the change, display it correctly
![image](https://user-images.githubusercontent.com/40540394/208619326-43bdb721-494f-4bc2-b2ac-f69c1d7a6132.png)

Reply:
Should you maybe add a test case that covers this scenario since it points out that a future maintainer might make the same mistake and we wouldn't have a current unit test that catches it.